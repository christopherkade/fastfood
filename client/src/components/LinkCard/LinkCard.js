import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

const Wrapper = styled.div`
  padding: 1rem;
  background-color: #EF798A;
  text-align: center;
  border-radius: 4px;
  margin-top: 0.3125rem;
`

const LinkText = styled.p`
  color: white;
  margin: 0 0 1rem 0;
`

const Link = styled.input`
  background-color: white;
  font-size: 1rem;
  border-radius: 4px;
  border: 1px solid white;
  padding: 0.25rem;
  width: 350px;

  &:focus {
    outline: transparent;
  }
`

const LinkCard = ({ link }) => {
  return (
    <Wrapper>
      <LinkText>Share this link with your friends!</LinkText>
      <Link value={link} readOnly />
    </Wrapper>
  )
}

LinkCard.propTypes = {
  link: PropTypes.string.isRequired
}

export default LinkCard
