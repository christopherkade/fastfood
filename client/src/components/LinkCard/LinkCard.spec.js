import React from 'react'
import { render, cleanup } from 'react-testing-library'

import LinkCard from "./LinkCard"

afterEach(cleanup);

describe("<LinkCard />", () => {
  describe("Snapshots", () => {
    it("should match snapshot", () => {
      const { container } = render(<LinkCard link="mockedLink" />);
      expect(container).toMatchSnapshot();
    });
  })
})
