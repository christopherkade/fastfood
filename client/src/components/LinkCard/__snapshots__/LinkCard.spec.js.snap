// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkCard /> Snapshots should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa ctMcSt"
  >
    <p
      class="sc-bwzfXH ePLGZB"
    >
      Share this link with your friends!
    </p>
    <input
      class="sc-htpNat FgSYl"
      readonly=""
      value="mockedLink"
    />
  </div>
</div>
`;
